src/lib/DepositApiClient.js

```diff
@@ -97,12 +97,20 @@ export class RDMDepositApiClient extends DepositApiClient {
       );
       return new DepositApiClientResponse(data, errors);
     } catch (error) {
-      const data = this.recordSerializer.deserialize(error.response.data || {});
+      const responseData = error.response.data;
+      const data = this.recordSerializer.deserialize(responseData || {});
+
+      const noFieldErrors = !('errors' in responseData);
+
+      if (noFieldErrors) {
+        throw new DepositApiClientResponse(data, responseData);
+      }
+
```

Comment:
suggestion: this method could be rewritten in 12 lines with a simplified logic:

```javascript
async _createResponse(axiosRequest) {
    let responseData = {}
    try {
        const response = await axiosRequest();
        responseData = response.data;
    } catch (error) {
        responseData = error.response.data;
    }
    const data = this.recordSerializer.deserialize(responseData || {});
    const errors = this.recordSerializer.deserializeErrors(
        response.data.errors || []
    );
    return new DepositApiClientResponse(data, errors);
}
```

Reply:
i tried it but then i realized that it either needs to return or throw the final value otherwise it will not work correctly

Reply:
In that case adding a boolean flag would be enough.

```javascript
let isThrowing = false;

try {
    ...
}
catch{
  ... 
  isThrowing = true;
}

if (isThrowing) {
    throw new ...
}

return ...
```

Note that this is a suggestion on code readability, it is not an optimization of any sorts so it is completely a suggestion that can be ignored if you want :)

Reply:
thank you!